cocos/particle/billboard.ts

```diff
@@ -113,6 +113,54 @@ export class Billboard extends Component {
         }
     }
 
+    @serializable
+    private _techIndex = 0;
+
+    @tooltip('i18n:billboard.technique')
+    public get technique (): number {
+        return this._techIndex;
+    }
+
+    public set technique (val: number) {
+        // clamp technique index
+        val = Math.floor(val);
+        if (val < 0) {
+            val = 0;
+        }
+        const techs = this._material?.effectAsset?.techniques;
+        if (techs && val >= techs.length) {
+            val = techs.length - 1;
+        }
+        // set technique index
+        this._techIndex = val;
+
+        // recreate model
+        if (this._model && this._mesh && this._material && this._material.technique !== val) {
+            // destroy model
+            this.detachFromScene();
+            this._model.destroy();
+            this._model = null;
+            this._material.destroy();
+            this._material = null;
+            this._mesh.destroy();
+            this._mesh = null;
+            // recreate model
+            this.createModel();
+            // set properties
+            this.width = this._width;
+            this.height = this._height;
+            this.rotation = this.rotation;
+            this.texture = this.texture;
```

Comment:
It seems strange to set properties like this.

Reply:
Refactored these assignments